editor/plugins/parallax_background_editor_plugin.cpp

```diff
@@ -0,0 +1,145 @@
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "parallax_background_editor_plugin.h"
+
+#include "canvas_item_editor_plugin.h"
+#include "editor/editor_node.h"
+#include "editor/editor_undo_redo_manager.h"
+#include "editor/scene_tree_dock.h"
+#include "scene/2d/parallax_2d.h"
+#include "scene/2d/parallax_layer.h"
+#include "scene/gui/menu_button.h"
+#include "scene/gui/spin_box.h"
+
+void ParallaxBackgroundEditorPlugin::edit(Object *p_object) {
+	parallax_background = Object::cast_to<ParallaxBackground>(p_object);
+}
+
+bool ParallaxBackgroundEditorPlugin::handles(Object *p_object) const {
+	return p_object->is_class("ParallaxBackground");
+}
+
+void ParallaxBackgroundEditorPlugin::make_visible(bool p_visible) {
+	if (p_visible) {
+		toolbar->show();
+	} else {
+		toolbar->hide();
+	}
+}
+
+void ParallaxBackgroundEditorPlugin::_menu_callback(int p_idx) {
+	switch (p_idx) {
+		case MENU_CONVERT_TO_PARALLAX_2D: {
+			convert_to_parallax2d();
+		} break;
+	}
+}
+
+void ParallaxBackgroundEditorPlugin::convert_to_parallax2d() {
+	ParallaxBackground *parallax_bg = parallax_background;
```

Comment:
What is the new pointer for?

Reply:
This is necessary because calling `replace_node()` internally focuses the editor UI on the newly replaced node, meaning the the `parallax_background` variable will be overwritten while the process is in play.

Reply:
Ah, right.